xc/device.ml

```diff
@@ -605,8 +605,9 @@ let qemu_media_change ~xs device _type params =
 	Xs.transaction xs (fun t -> t.Xst.writev backend_path back_delta);
 	debug "Media changed: params = %s" pathtowrite;
 
-	try
-		ignore(with_xs (fun xs -> xs.Xs.read (Printf.sprintf "/libxl/%d/dm-version" device.frontend.domid)) = "qemu_xen");
+
+	if is_upstream_qemu device.frontend.domid
+	then begin
 		let c = Qmp_protocol.connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" device.frontend.domid) in
```

Comment:
And I think generally only one QMP message per time when communicating with QEMU.
So it's not necessary to put message in a list.